src/gwt/src/org/rstudio/core/client/widget/TextBoxWithButton.java

```diff
@@ -86,6 +87,11 @@ protected TextBoxWithButton(String label,
       textBox_ = new TextBox();
       textBox_.setWidth("100%");
       textBox_.setReadOnly(readOnly);
+      
+      textBox_.addValueChangeHandler((ValueChangeEvent<String> event) ->
```

Comment:
The changes here (and below) were made to ensure that value change events fire when the underlying textbox itself is edited, which is necessary for the case where the textbox is not read-only.